testing/test_cacheprovider.py

```diff
@@ -603,3 +603,116 @@ def test_foo_4():
         result = testdir.runpytest('--last-failed')
         result.stdout.fnmatch_lines('*4 passed*')
         assert self.get_cached_last_failed(testdir) == []
+
+
+class TestNewFirst(object):
+    def test_newfirst_usecase(self, testdir):
+        t1 = testdir.mkdir("test_1")
+        t2 = testdir.mkdir("test_2")
+
+        t1.join("test_1.py").write(
+            "def test_1(): assert 1\n"
+            "def test_2(): assert 1\n"
+            "def test_3(): assert 1\n"
+        )
+        t2.join("test_2.py").write(
+            "def test_1(): assert 1\n"
+            "def test_2(): assert 1\n"
+            "def test_3(): assert 1\n"
+        )
+
+        path_to_test_1 = str('{}/test_1/test_1.py'.format(testdir.tmpdir))
+        os.utime(path_to_test_1, (1, 1))
```

Comment:
You can use [setmtime](http://py.readthedocs.io/en/latest/path.html#py._path.local.LocalPath.setmtime) here for the same effect.

```python
testdir.tmpdir.join('test_1/test_1.py').setmtime(1)
```